VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts

```diff
@@ -154,6 +160,15 @@ export abstract class WaitingRoomBaseDirective {
         return str.replace(/\s/g, '-').toLowerCase();
     }
 
+    togglePanel(panelName: string) {
+        const newState = !this.panelStates[panelName];
+        this.panelTypes.forEach(pt => {
+            this.panelStates[pt] = false;
+        });
```

Comment:
The if statement that you removed could be added back here with the changes to the tests.